src/main/latex/paper/sections/tools/errorprone.tex

```diff
@@ -0,0 +1,51 @@
+%! Author = florian
+%! Date = 6/23/23
+\subsection{ErrorProne}\label{subsec:errorprone}
+Error Prone is a popular Java library that aims to improve the quality and reliability of Java code by detecting and preventing common programming errors.
+It functions as a static analysis tool and compiler plugin, seamlessly integrating with the Java compiler to provide valuable feedback and suggestions during the development process.
+
+Issues are flagged by creating compiler errors with an appropriate message, which contextualizes errors and often include a possible fix.
+Similar to how modern IDEs often suggest a possible fix, before even compiling the source code.
+Error Prone is also able to create patch files with these fixes, that then can be applied by the developers.
+
+Error Prone can be extended by writing plugins.
+The plugins operate directly on the abstract syntax tree, which helps writing powerful plugins.
+The downside of this approach is that writing plugins is not an easy task.
+
+Refaster is a tool that aims to solve this problem.
+It provides an API with before and after templates.
+That way it is easy to create customized plugins for Error Prone.
+
+A similar tool to Error Prone is Spotbugs, which is described in the Subsection\ \ref{subsec:spotbugs}.
+In comparison to Spotbugs, Error Prone offers a more powerful configuration and plugin system, making Error Prone the preferred static analysis tool~\cite{errorprone}.
+
+\subsubsection{Test Setup}
+Error Prone is easy to integrate in existing gradle projects.
+Similar to Checkstyle, which is described in Subsection\ \ref{subsec:checkstyle}.
+The plugin simply has to be included, and it will integrate itself into the \texttt{compileJava}-task.
+The Figure\ \ref{fig:errorprone-gradle} shows the configuration in the \texttt{build.gradle}.
+
+\begin{figure}[h]
+    \caption{Checkstyle build.gradle}
+    \lstinputlisting[basicstyle=\tiny,label={lst:errorprone-gradle}]{../../errorprone/build.gradle}
+    \label{fig:errorprone-gradle}
+\end{figure}
+
+\subsubsection{Nullaway}
+A plugin which should be mentioned is Nullaway.
+It helps developers finding possible NullPointerExceptions in the code, which is in Java still a problem, since there is no real option type, like in \.Net or TypeScript.
```

Comment:
```suggestion
It helps developers finding possible NullPointerExceptions in the code, which is still a problem in Java, since there is no real option type, like in \.Net or TypeScript.
```